x-pack/legacy/plugins/lens/public/app_plugin/app.test.tsx

```diff
@@ -114,6 +116,38 @@ describe('Lens App', () => {
     `);
   });
 
+  describe('setBreadcrumbs', () => {
+    it('contains only Visualize if no document is passed', () => {
+      const mockSet = jest.fn();
+      setBreadcrumbs({
+        addBasePath: (s: string) => `/testbasepath${s}`,
+        breadcrumbs: { set: mockSet },
+      });
+
+      expect(mockSet).toHaveBeenCalledWith([
+        { text: 'Visualize', href: '/testbasepath/app/kibana#/visualize' },
+      ]);
+    });
+
+    it('shows the document title if a document is passed', () => {
+      const mockSet = jest.fn();
+      setBreadcrumbs(
+        {
+          addBasePath: (s: string) => `/testbasepath${s}`,
+          breadcrumbs: { set: mockSet },
+        },
+        {
+          title: 'If you want it done, go. If not, send.',
+        }
+      );
+
+      expect(mockSet).toHaveBeenCalledWith([
+        { text: 'Visualize', href: '/testbasepath/app/kibana#/visualize' },
+        { text: 'If you want it done, go. If not, send.' },
+      ]);
+    });
+  });
```

Comment:
I don't see a test for the useEffect you added, which would be if the persisted title is changed, the breadcrumb should change.

Reply:
Yeah... It's true. It's clumsy to test, and I felt like I was just testing mocks, mostly, but it's probably worth something basic at least.